### PR TITLE
[wasm][debugger] Fix concurrent access to cache.

### PR DESCRIPTION
**Pull request by thaystg**
PropertiesBinaryReader could be access by more than one thread which was causing starting reading from the wrong position and skip positions.
Added a test case for it.
@lewing I would like to backport to rc2.

### CODE REVIEW

**Review by lewing — APPROVED** [review]

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs [hunk 1]
```diff
@@ -1212,18 +1212,23 @@ public async Task<MonoBinaryReader> GetTypePropertiesReader(SessionId sessionId,
...
             var commandParams = new MemoryStream();
             var commandParamsWriter = new MonoBinaryWriter(commandParams);
             commandParamsWriter.Write(typeId);
 
-            typeInfo.PropertiesBinaryReader = await SendDebuggerAgentCommand<CmdType>(sessionId, CmdType.GetProperties, commandParams, token);
-            return typeInfo.PropertiesBinaryReader;
+            var reader = await SendDebuggerAgentCommand<CmdType>(sessionId, CmdType.GetProperties, commandParams, token);
+            var ms = new MemoryStream();
+            reader.BaseStream.CopyTo(ms);
+            typeInfo.PropertiesBuffer = ms.ToArray();
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:1229]: Would it make sense to use `typeInfo.PropertiesBuffer = reader.ReadBytes(reader.BaseStream.Length);`?

File: src/mono/wasm/debugger/DebuggerTestSuite/CustomViewTests.cs [hunk 2]
```diff
@@ -64,5 +64,40 @@ public async Task UsingDebuggerTypeProxy()
...
+            var pause_location = await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_static_method ('[debugger-test] DebuggerTests.DebuggerCustomViewTest2:run'); }, 1);",
+                "dotnet://debugger-test.dll/debugger-custom-view-test.cs",
+                bp.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp.Value["locations"][0]["columnNumber"].Value<int>(),
+                "run");
+
+            pause_location = await StepAndCheck(StepKind.Over, "dotnet://debugger-test.dll/debugger-custom-view-test.cs", bp.Value["locations"][0]["lineNumber"].Value<int>()+2, bp.Value["locations"][0]["columnNumber"].Value<int>(),  "run");
+
+            System.Collections.Generic.List<Task<bool>> tasks = new System.Collections.Generic.List<Task<bool>>();
```
Comment (radical) [inline src/mono/wasm/debugger/DebuggerTestSuite/CustomViewTests.cs:91]: Add a `using System.Collections.Generic;`, and change this to `List<Task<bool>> tasks = new();`

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs (hunk 1 above)
Comment (thaystg) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:1229]: I think I need this cast to work: `typeInfo.PropertiesBuffer = reader.ReadBytes((int)reader.BaseStream.Length);`
Are you okay with that?

File: src/mono/wasm/debugger/DebuggerTestSuite/CustomViewTests.cs (hunk 2 above)
Comment (thaystg) [reply src/mono/wasm/debugger/DebuggerTestSuite/CustomViewTests.cs:91]: done

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs (hunk 1 above)
Comment (thaystg) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs]: done

**Review by radical — APPROVED** [review]
LGTM!

**Pull request merged**